app/controllers/application.js

```diff
@@ -1,57 +1,20 @@
 import Controller, { inject as controller } from '@ember/controller';
 import { oneWay } from '@ember/object/computed';
-import $ from 'jquery';
+import { EKMixin, keyDown, keyPress } from 'ember-keyboard';
+import { on } from '@ember/object/evented';
 
-export default Controller.extend({
+export default Controller.extend(EKMixin, {
     searchController: controller('search'),
     search: oneWay('searchController.q'),
 
-    init() {
-        this._super(...arguments);
-        $(document).on('keypress', this.handleKeyPress.bind(this));
-        $(document).on('keydown', this.handleKeyPress.bind(this));
-    },
-
-    // Gets the human-readable string for the virtual-key code of the
-    // given KeyboardEvent, ev.
-    //
-    // This function is meant as a polyfill for KeyboardEvent#key,
-    // since it is not supported in Trident.  We also test for
-    // KeyboardEvent#keyCode because the handleShortcut handler is
-    // also registered for the keydown event, because Blink doesn't fire
-    // keypress on hitting the Escape key.
-    //
-    // So I guess you could say things are getting pretty interoperable.
-    getVirtualKey(ev) {
-        if ('key' in ev && typeof ev.key !== 'undefined') {
-            return ev.key;
-        }
-        const c = ev.charCode || ev.keyCode;
-        if (c === 27) {
-            return 'Escape';
-        }
-        return String.fromCharCode(c);
-    },
-
-    handleKeyPress(evt) {
-        // Don't focus the search field if the user is already using an input element
-        if (evt.target.tagName === 'INPUT' || evt.target.tagName === 'TEXTAREA' || evt.target.tagName === 'SELECT') {
+    keyboardActivated: true,
+    focusSearch: on(keyDown('KeyS'), keyPress('KeyS'), function(event) {
+        if (event.ctrlKey || event.altKey || event.metaKey) {
             return;
         }
-        // Only match plain keys, no modifiers keys
-        if (evt.ctrlKey || evt.altKey || evt.metaKey) {
-            return;
-        }
-        if (this.getVirtualKey(evt).toLowerCase() === 's') {
-            evt.preventDefault();
-            $('#cargo-desktop-search').focus();
-        }
-    },
-
-    willDestroy() {
-        $(document).off('keypress');
-        $(document).off('keydown');
-    },
+        event.preventDefault();
+        document.querySelector('#cargo-desktop-search').focus();
```

Comment:
can you check if that also works on small screens? I roughly remember that the search field is not always shown depending on screen size.

Reply:
You right, it doesn't work on small (mobile) screen. This is not new - it was already not working before my change. But on a small screen (phone, tablet) you usually do not use keyboard shortcuts. I could test if "desktop" input is visible and focus accordingly, but I am not sure if it worth it.

Reply:
I think `$('#cargo-desktop-search').focus()` would work since it will just noop if the selector has no matches.

```js
let searchField = document.querySelector('#cargo-desktop-search');
if (searchField) {
  searchField.focus();
}
```

How about this?

Reply:
nevermind, I just noticed that this is not necessary

Reply:
Yes, I added your change but I could not reproduce the error...

Reply:
I guess better safe then sorry :)